roles/ceph-osd/tasks/check_devices_static.yml

```diff
@@ -1,4 +1,9 @@
 ---
+- name: make sure the devices variable is a list
+  fail:
+    msg: "devices: must be a list, not a string, i.e. [ \"/dev/sda\" ]"
+  when: devices[0] | length == 1
```

Comment:
I think a safer check here might be: ``when: devices is string``.

Could you try that out and see if it works?